File: lua/Decimal-Binary.lua
-- c 2017 Michael Smith

local decimal = io.read()
local binary = "";
local bytes = { 128, 64, 32, 16, 8, 4, 2, 1 }
while decimal > 0 do
    for i = 0, i < bytes.length, i + 1 do
        if decimal < 256 then
            if decimal >= bytes[i] then
                decimal = decimal - bytes[i]
                binary = binary .. "" .. 1 .. ""
            else
                binary = binary .. "" .. 0 .. ""
            end
        else
            decimal = decimal - 255
            binary = binary .. "" .. 11111111 .. " "
        end
    end
    io.write(binary)
end

File: java/Decimal-Binary.java
import java.io.*;
import static java.lang.System.*;

import java.util.Scanner;

class Main{

     public static void main (String str[]) throws IOException {

         // c 2017 Michael Smith
         
         Scanner scan = new Scanner(System.in);
         int decimal = scan.nextInt();
         String binary = "";
         int[] bits = { 128, 64, 32, 16, 8, 4, 2, 1 };
         while ( decimal > 0 ) {
         for (int i = 0; i < bytes.length; i++) {
           if (decimal < 256) {
             if (decimal >= bits[i]) {
               decimal -= bits[i];
                 binary += "" + 1 + "";
             } else {
                 binary +="" + 0 + "";
             }
           } else {
               decimal -= 255;
               binary += "" + 11111111 + " ";
           }
         }
         System.out.println(binary);
       }
       
     }

}
